drupal/web/modules/custom/covid/src/Normalizer.php

```diff
@@ -1,27 +1,45 @@
 <?php
 
-
 namespace Drupal\covid;
 
-
+use Drupal\Core\Field\Plugin\Field\FieldType\StringLongItem;
 use Drupal\Core\TypedData\Plugin\DataType\StringData;
 use Drupal\link\Plugin\Field\FieldType\LinkItem;
 use Drupal\serialization\Normalizer\TypedDataNormalizer;
-
+use Drupal\text\Plugin\Field\FieldType\TextLongItem;
+use Drupal\text\TextProcessed;
+
+/**
+ * Class Normalizer.
+ *
+ * @package Drupal\covid
+ */
 class Normalizer extends TypedDataNormalizer {
 
+  /**
+   * {@inheritdoc}
+   */
   public function normalize($object, $format = NULL, array $context = []) {
     $data = parent::normalize($object, $format, $context);
-
-    if ($object instanceof StringData && ($parent = $object->getParent()) instanceof LinkItem && $object->getString() === "") {
-      $parsedUrl = parse_url($parent->get('uri')->getValue());
-      $url = '';
-      foreach (['host', 'path'] as $part) {
-        if (isset($parsedUrl[$part])) {
-          $url .= $parsedUrl[$part];
+    $parent = $object->getParent();
+    if ( in_array(get_class($object), [StringData::class, StringLongItem::class])) {
+      if ($parent instanceof LinkItem && $object->getString() === "") {
+        $parsedUrl = parse_url($parent->get('uri')->getValue());
+        $url = '';
+        foreach (['host', 'path'] as $part) {
+          if (isset($parsedUrl[$part])) {
+            $url .= $parsedUrl[$part];
+          }
         }
+        return $url;
       }
-      return $url;
+      $space = html_entity_decode("&nbsp;");
+      return mb_ereg_replace('(?<![<-])\b(\w(?:\b|\.?))(<? )', "\\1$space", $object->getString());
+    }
+    elseif ($object instanceof TextProcessed && $parent instanceof TextLongItem && $object->getName() === 'processed') {
```

Comment:
Taková blbá otázka, za kterou mě asi nebudeš mít rád, ale nebylo by lepší tohle řešit spíš jako text filter pro ckeditor? :)

Reply:
nad tym som rozmyslal a zdalo sa mi lepsie mat to iba pre api vystup. rovnako sa filter neaplikuje na plain text fieldy, ktore je tu potrebne riesit, takze by som mal bud duplikovany kod alebo by som tu musel vyrobit instanciu filtra a pouzit ho, co mi zase pripada viac resource-heavy.

Reply:
ok to asi dava smysl.